src/Generating/ChunkDesc.cpp

```diff
@@ -587,21 +587,21 @@ cBlockEntity * cChunkDesc::GetBlockEntity(int a_RelX, int a_RelY, int a_RelZ)
 		else
 		{
 			// Wrong type, the block type has been overwritten. Erase and create new:
-			m_BlockEntities.erase(itr);
+			m_BlockArea.GetBlockEntities().erase(Iterator);
 		}
 	}
 
 	int AbsX = a_RelX + m_Coords.m_ChunkX * cChunkDef::Width;
 	int AbsZ = a_RelZ + m_Coords.m_ChunkZ * cChunkDef::Width;
 
 	// The block entity is not created yet, try to create it and add to list:
-	auto be = cBlockEntity::CreateByBlockType(GetBlockType(a_RelX, a_RelY, a_RelZ), GetBlockMeta(a_RelX, a_RelY, a_RelZ), {AbsX, a_RelY, AbsZ});
-	if (be == nullptr)
+	auto BlockEntity = cBlockEntity::CreateByBlockType(GetBlockType(a_RelX, a_RelY, a_RelZ), GetBlockMeta(a_RelX, a_RelY, a_RelZ), {AbsX, a_RelY, AbsZ});
+	if (BlockEntity == nullptr)
 	{
 		// No block entity for this block type
 		return nullptr;
 	}
-	auto res = m_BlockEntities.emplace(Idx, std::move(be));
+	auto res = m_BlockArea.GetBlockEntities().emplace(Idx, std::move(BlockEntity));
```

Comment:
Does BlockArea not keep track of its internal blockentities list correctly?

Reply:
No. When using the prefab the positions are not set correctly.

The relative position is correct but the absolute is wrong. This only happens when using a prefab. This code is only a bad fix - I didn't find the underlying issue.

The problem is that a block entity is not informed by the world where it's actual position is. The broadcasting, loading, saving is done with the position stored in the block entity.